pandas/tests/series/test_constructors.py

```diff
@@ -688,10 +688,21 @@ def test_constructor_dtype_datetime64(self):
         pytest.raises(TypeError, lambda x: Series(
             Series(dates).astype('int') / 1000000, dtype='M8[ms]'))
 
-        msg = (r"The 'datetime64' dtype has no unit\. Please pass in"
```

Comment:
this is explicitly tested by `test_astype_generic_timestamp_no_frequency`. however i've now updated this test so that the error message check is more explicit.

Reply:
@WillAyd 

i'm guessing that the intended error message for this test would have been originally that raised by
https://github.com/pandas-dev/pandas/blob/af82b2a3fe202d1ee5691e33e0a7807e6f7499f0/pandas/core/dtypes/cast.py#L644-L646
and before the conversion to pytest.raises context manager the test was
https://github.com/pandas-dev/pandas/blob/17a6bc56e5ab6ad3dab12d3a8b20ed69a5830b6f/pandas/tests/series/test_constructors.py#L680-L681

which would have been raising `TypeError: missing 1 required positional argument: 'x'`

since there was no message check the tested function raising `TypeError` would have incorrectly been considered a test pass

so this test was never intended to test for `The 'datetime64' dtype has no unit` message.

hope that makes it clear!

Reply:
so I DO want this test to remain, can you simply change it so it is testing with ``dtype='datetime64'``, IOW it raises.

Reply:
This *might* actually be a duplicate test (if so, then can remove)

Reply:
ok, you are going to remove this test?

Reply:
not removing... since it now passes, additionally constructed the expected result and compared.

Reply:
> ok, you are going to remove this test? rather, this case?

so have now added some cases to test_constructor_dtype_datetime64 similar to those in test_constructor_dtype_timedelta64 and also added the exception message check to tests/frame/test_dtypes.py

so could probably remove some of the # GH3414 related stuff